### PR TITLE
Feature/content steering

### PR DESCRIPTION
**Pull request by fernandocQualabs**
## Description
This is the first step integrating the Content Steering functionality in CML. 
We added fetching function to Content Steering server to CML

### CODE REVIEW

File: lib/src/contentsteering/contentSteering.ts
Comment (mpollingerQualabs) [inline lib/src/contentsteering/contentSteering.ts:1]: Is this a test file?

File: lib/src/contentsteering/contentSteering.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const manifestUrl =
+	'https://fastly.content-steering.com/bbb_hls/master_steering_fastly_https.m3u8thisIsAWrongUri';
```
Comment (mpollingerQualabs) [inline lib/src/contentsteering/contentSteering.ts]: Should this be here?

Comment (fernandocQualabs) [reply lib/src/contentsteering/contentSteering.ts]: i commited a test file, my apologies

File: lib/src/contentsteering/callContentSteeringServer.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	headers: Record<string, string>;
+	url: string;
+}
+
+export async function callContentSteeringServer(
+	serverURI: string | null,
+	onSuccess: (response: ContentSteeringServerResponse) => void,
+	onError: (response: ContentSteeringServerResponse) => void,
+	onTimeout: (serverURI: string) => void,
+	timeout: number = 5000
```
Comment (fpyoung) [inline lib/src/contentsteering/callContentSteeringServer.ts:13]: Maybe these functions should come in an object named callbacks just like it's implemented in hls.js:
Screenshot 2024-03-21 at 9 26 11 AM
On the other hand in dash it also comes inside an object: 
image
What do you think?

File: lib/src/contentsteering/callContentSteeringServer.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		};
+
+		if (response.status >= 200 && response.status < 300) {
+			onSuccess(ContentSteeringServerResponse);
+		}
+		else if (response.status >= 400 && response.status < 600) {
+			onError(ContentSteeringServerResponse);
+		}
+	}
+	catch (error: any) {
```
Comment (fpyoung) [inline lib/src/contentsteering/callContentSteeringServer.ts:44]: Can we remove this any from here?

File: lib/src/contentsteering/parseM3u8ContentSteeringURI.ts
Comment (fpyoung) [inline lib/src/contentsteering/parseM3u8ContentSteeringURI.ts:1]: this function isn't part of the just-fetching logic, maybe we should remove it

**Review by fpyoung — CHANGES_REQUESTED** [review]

**Review by coatesjuan — CHANGES_REQUESTED** [review]
resolve comments

File: lib/src/contentsteering/parseM3u8ContentSteeringURI.ts
Comment (fpyoung) [inline lib/src/contentsteering/parseM3u8ContentSteeringURI.ts:1]: We used this function for testing purposes, I think this function is really useful but maybe should be placed somewhere else.

**Review by fpyoung — APPROVED** [review]

**Review by coatesjuan — APPROVED** [review]

File: lib/src/contentsteering/contentSteering.ts
Comment (fpyoung) [reply lib/src/contentsteering/contentSteering.ts:1]: This file was removed, it was only created for testing purposes.

File: lib/src/contentsteering/callContentSteeringServer.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	headers: Record<string, string>;
+	url: string;
+}
+
+export async function callContentSteeringServer(
+	serverURI: string | null,
+	onSuccess: (response: ContentSteeringServerResponse) => void,
+	onError: (response: ContentSteeringServerResponse) => void,
+	onTimeout: (serverURI: string) => void,
+	timeout: number = 5000
```
Comment (mpollingerQualabs) [inline lib/src/contentsteering/callContentSteeringServer.ts:13]: I would add a comment to this value specifying it's milliseconds. Or change the name of the variable to `timeoutMilliseconds` or something like that, just for clarity and avoiding confusion.
If want to go an extra step, even make it a constant at the top.

File: lib/src/contentsteering/callContentSteeringServer.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		};
+
+		if (response.status >= 200 && response.status < 300) {
+			onSuccess(ContentSteeringServerResponse);
+		}
+		else if (response.status >= 400 && response.status < 600) {
+			onError(ContentSteeringServerResponse);
+		}
+	}
+	catch (error: any) {
```
Comment (mpollingerQualabs) [inline lib/src/contentsteering/callContentSteeringServer.ts:44]: Can this be anything other than `any`?

**Review by mpollingerQualabs — CHANGES_REQUESTED** [review]

File: lib/src/contentsteering/callContentSteeringServer.ts (hunk 4 above)
Comment (mpollingerQualabs) [reply lib/src/contentsteering/callContentSteeringServer.ts:13]: Btw, i've assumed this is in milliseconds...

**Pull request closed without merging**